### PR TITLE
Add check for empty time spans

### PR DESCRIPTION
**Pull request by pirkkahuhtala**

### CODE REVIEW

File: hours/models.py [hunk 1]
```diff
@@ -667,6 +667,9 @@ def as_hash_input(self) -> str:
     def as_text(self) -> str:
         group_strings = []
         for time_span_group in self.time_span_groups.all():
+            # we don't know why there are empty time_spans vector, this is a workaround for the time being
+            if len (time_span_group.time_spans.all()) == 0:
+                continue
             if time_span_group.is_removed:
                 continue
```
Comment (mikkokeskinen) [inline hours/models.py:674]: I would first check if the group is removed, only then check if the group has any time spans.

Also in Django the correct way to check if there are items in a queryset is with count. i.e.:

```python
if not time_span_group.time_spans.count():
```

Commit message could be changed too... We know that there can be empty time span groups. If someone removes all time spans from a group the group is not automatically removed.

**Review by mikkokeskinen — CHANGES_REQUESTED** [review]

File: hours/models.py (hunk 1 above)
Comment (pirkkahuhtala) [reply hours/models.py:674]: Oh nice! Thanks. I'll make the changes.

**Comment by mikkokeskinen** [comment]
You could have just amended the old commit. No need to keep the changes in the commit history.

Now you should squash those commits to one.

**Comment by pirkkahuhtala** [comment]
> You could have just amended the old commit. No need to keep the changes in the commit history.
> 
> Now you should squash those commits to one.

Alright. I'll squash em. Unsure about the convention in this repo. Thanks.

**Review by mikkokeskinen — APPROVED** [review]

**Pull request merged**